app/viam_client.go

```diff
@@ -49,16 +50,30 @@ func CreateViamClientWithOptions(ctx context.Context, options Options, logger lo
 		return nil, err
 	}
 
-	if options.Credentials.Payload == "" || options.Entity == "" {
-		return nil, errors.New("entity and payload cannot be empty")
+	var conn rpc.ClientConn
+	if len(options.DialOptions) > 0 {
+		conn, err = dialDirectGRPC(ctx, serviceHost.Host, logger, options.DialOptions...)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		if options.Credentials.Payload == "" || options.Entity == "" {
+			return nil, errors.New("entity and payload cannot be empty")
+		}
+		opts := rpc.WithEntityCredentials(options.Entity, options.Credentials)
+		conn, err = dialDirectGRPC(ctx, serviceHost.Host, logger, opts)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
not sure if this is an anti pattern/ something we'd want to avoid. if DialOptions exists im ignoring anythign else set in options except baseUrl. 

[RobotClient.New](https://github.com/jr22/rdk/blob/main/robot/client/client.go#L264-L266) is modeled totally differently and so are [RobotClientOptions](https://github.com/jr22/rdk/blob/main/robot/client/client_options.go#L49-L51) so not sure if its preferable I handle this logic in a different way